internal/provider/retry_predicates_test.go

```diff
@@ -56,3 +56,31 @@ func TestIsOperationReadQuotaError_rateLimitExceeded(t *testing.T) {
 		t.Errorf("Error not detected as retryable")
 	}
 }
+
+func TestGoogle404Error(t *testing.T) {
+	gerr := googleapi.Error{
+		Code:    404,
+		Message: "notfound",
+	}
+	err := &gerr
+
+	expected := true
+
+	if isNotFound(err) != expected {
+		t.Error("Failed: The error should have been detected as 404")
+	}
+}
+
+func TestGoogleNot404Error(t *testing.T) {
+	gerr := googleapi.Error{
+		Code:    200,
+		Message: "notfound",
+	}
+	err := &gerr
+
+	expected := false
+
+	if isNotFound(err) != expected {
+		t.Error("Failed: The error should have been detected as 404")
+	}
```

Comment:
Looks like something overlooked from when some copy/pasting was done- if this test failed the error message would be a bit confusing to developers initially

Reply:
Hi @SarahFrench sorry for the faux pas and sorry it took me so long to fix it.
I changed the error message and made it totaly different than the other bool test (nothing worse than missing a little not in an error message during tests)